NewForce-Cohort-5/tabloidmvc-the-daily-debuggers
# View post comments np

# Description

I added a comments section to the posts. While viewing a post details a user can click on the "Comments Button." A user can view all comments. There is also a button which will return the user back to the post once they are finished viewing the comments.


## Type of change

[Check Trello Ticket #19](https://trello.com/b/ECtaPfZr/the-daily-debuggers)


# Testing Instructions

1. git add and git commit all changes in your current branch.
2. git checkout main
3. git fetch -a
4. git checkout view-post-comments-np
5. Navigate to the root directory for this repo (directory-to)/tabloidmvc-the-daily-debuggers
6. Start TabloidMVC.sln
7. Make sure your db is created with seed data (Skip to 10 if this is already done)
8. Run [01_TabloidMVC_Create_DB.sql](https://github.com/NewForce-Cohort-5/tabloidmvc-the-daily-debuggers/blob/main/SQL/01_TabloidMVC_Create_DB.sql) and [02_TabloidMVC_Seed_Data.sql](https://github.com/NewForce-Cohort-5/tabloidmvc-the-daily-debuggers/blob/main/SQL/02_TabloidMVC_Seed_Data.sql) queries in Visual Studio
9. Run query below to add seed data for comments
10. Start the server in Debug Mode F5
11. Log in as admin admin@example.com (or any user)
12. Navigate to the posts link
13. Navigate to post details
14. Click on comments - user should be able to see comments related to post as well as a "Back to Post" button
15. Click on "Back to Post" to return to viewing the post

Use the following SQL query to add Comment seed data
```sql
INSERT INTO Comment (PostId, UserProfileId, [Subject], Content, CreateDateTime)
Values (1, 2, 'Really good', 'I really enjoyed reading this article', GETDATE())

INSERT INTO Comment (PostId, UserProfileId, [Subject], Content, CreateDateTime)
Values (1, 1, 'WoW', 'What an amazing read!', GETDATE())
```

Use the following SQL query to see what you should expect to see on the website
```sql
SELECT * FROM Comment

```
Use the following SQL query to add users that are not admins if needed
```sql
INSERT INTO UserProfile (DisplayName, FirstName, LastName, Email, CreateDateTime, ImageLocation, UserTypeId)
VALUES ('Andy', 'Andy', 'Birt', 'abirt@mail.com', GETDATE(), null, 2)

INSERT INTO UserProfile (DisplayName, FirstName, LastName, Email, CreateDateTime, ImageLocation, UserTypeId)
VALUES ('Nick', 'Nick', 'Perry', 'nperry@mail.com', GETDATE(), null, 2)

INSERT INTO UserProfile (DisplayName, FirstName, LastName, Email, CreateDateTime, ImageLocation, UserTypeId)
VALUES ('Brent', 'Brent', 'Griffith', 'bgriffith@mail.com', GETDATE(), null, 2)

```


# Checklist:
